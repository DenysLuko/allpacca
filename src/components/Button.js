import React, { Component } from 'react'
import {
  TouchableHighlight,
  StyleSheet,
  Text,
  View
} from 'react-native'
import { styles } from '../styles'

const buttonStyles = StyleSheet.create({
  homeButton: {
    height: 70,
    width: '70%',
    maxWidth: 300,
    borderColor: styles.colors.blue.light,
    borderWidth: 1,
    borderRadius: 50
  },
  textContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  textContent: {
    ...styles.text.headingText,
    color: styles.colors.grey.dark
  }
})

export const Button = ({
  title,
  onPress,
  type
}) => (
  <TouchableHighlight
    onPress={onPress}
    style={buttonStyles[type]}
    underlayColor={styles.colors.blue.lightest}
  >
    <View style={buttonStyles.textContainer}>
      <Text style={buttonStyles.textContent}>{ title }</Text>
    </View>
  </TouchableHighlight>
)
