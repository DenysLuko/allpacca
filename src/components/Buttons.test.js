/**
 * @format
 */

import 'react-native'
import React from 'react'
import { Button } from './Button'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders homeButton correctly', () => {
  const subject = renderer.create(<Button type={'homeButton'} title={'Button Title'}/>).toJSON()
  expect(subject).toMatchSnapshot()
})
