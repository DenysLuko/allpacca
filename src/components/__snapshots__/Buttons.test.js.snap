// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homeButton correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#4db8b8",
      "borderRadius": 50,
      "borderWidth": 1,
      "height": 70,
      "maxWidth": 300,
      "width": "70%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#006b6b",
          "fontFamily": "Avenir-Light",
          "fontSize": 24,
        }
      }
    >
      Button Title
    </Text>
  </View>
</View>
`;
